python/lsst/ap/association/mpSkyEphemerisQuery.py

```diff
@@ -0,0 +1,218 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Solar System Object Query to MPSky in place of a internal Rubin solar
+system object caching/retrieval code.
+
+Will compute the location for of known SSObjects within a visit. This code
+blocks on web requests, so should not be used as part of any real-time or
+time-sensitive system. Use in a larger pipeline at your own risk.
+"""
+
+__all__ = ["MPSkyEphemerisQueryConfig", "MPSkyEphemerisQueryTask"]
+
+
+import pandas as pd
+import numpy as np
+import pyarrow as pa
+import requests
+import sys
+
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+from lsst.utils.timer import timeMethod
+from lsst.geom import SpherePoint
+
+from lsst.pipe.base import PipelineTask, PipelineTaskConfig, PipelineTaskConnections
+import lsst.pipe.base.connectionTypes as connTypes
+
+# Enforce an error for unsafe column/array value setting in pandas.
+pd.options.mode.chained_assignment = 'raise'
+
+
+class MPSkyEphemerisQueryConnections(PipelineTaskConnections,
+                                     dimensions=("instrument",
+                                                 "group", "detector")):
+
+    predictedRegionTime = pipeBase.connectionTypes.Input(
+        doc="The predicted exposure region and time",
+        name="regionTimeInfo",
+        storageClass="RegionTimeInfo",
+        dimensions={"instrument", "group", "detector"},
+    )
+
+    ssObjects = connTypes.Output(
+        doc="MPSky-provided Solar System objects observable in this detector-visit",
+        name="preloaded_ssObjects",
+        storageClass="DataFrame",
+        dimensions=("instrument", "group", "detector"),
+    )
+
+
+class MPSkyEphemerisQueryConfig(
+        PipelineTaskConfig,
+        pipelineConnections=MPSkyEphemerisQueryConnections):
+    observerCode = pexConfig.Field(
+        dtype=str,
+        doc="IAU Minor Planet Center observer code for queries "
+            "(Rubin Obs./LSST default is X05)",
+        default='X05'
+    )
+    queryRadiusDegrees = pexConfig.Field(
+        dtype=float,
+        doc="On sky radius for Ephemeris cone search. Defaults "
+            "to the radius of Rubin Obs FoV in degrees",
+        default=1.75
+    )
```

Comment:
Turns out there's no good documentation about making config overrides. 😞 A quick from-memory summary:

* The files in `obs_*/config/` contain config settings that should be applied for all runs of that task on that instrument. They have higher priority than task defaults, but lower priority than in-pipeline overrides.
* The config file must have the same name as the task label in the pipeline (which is usually, including in your case, `_DefaultName`).
* Overrides directly under `obs_lsst/config/` apply to _all_ LSST instruments. Overrides in a subdirectory apply only to that instrument, and have higher priority than the general overrides.

Reply:
On second glance, something is very wrong here -- you don't actually _use_ `observerCode` when uploading to MPSky. Is this a future feature?

Reply:
We should use X05 for AuxTel for now. AuxTel getting an observatory code is blocked on association working, since the Minor Planet Center assigns observatory codes and they've asked for some NEO measurements. I don't see an obs_auxtel equivalent though -- should that be somewhere in obs_lsst?

Now that I look at it, I don't see a SkyBotEphemerisQuery.py override in obs_lsst at all. Should I follow the format from obs_subaru and obs_decam?

Reply:
Yes, observerCode is a future MPSky feature, but it currently just defaults to LSST (X05).

Reply:
> I don't see an obs_auxtel equivalent though -- should that be somewhere in obs_lsst?

Yes, that's what the subdirectories are for. I _think_ `LATISS` is the only one that's for AuxTel instead of Simonyi?

> Now that I look at it, I don't see a SkyBotEphemerisQuery.py override in obs_lsst at all. Should I follow the format from obs_subaru and obs_decam?

The [default for SkyBotEphemerisQuery was I11](https://github.com/lsst/ap_association/blob/main/python/lsst/ap/association/skyBotEphemerisQuery.py#L80), which was already for LSST. I'm not sure how I11 relates to X05.

Where you _do_ need an override (LATISS only?), then yes, following the existing format should be enough.

Reply:
> Yes, observerCode is a future MPSky feature, but it currently just defaults to LSST (X05).

Please add a note to the field docstring that it's not used yet.

This does raise the question of what to do for DECam and especially HSC. We could just run the task and accept that nothing will match (that's what `SkyBotEphemerisQueryTask` did before I learned that geocentric is _not_ a good default 😛), or we could trim the task out of those pipelines until MPSky supports them.

Reply:
I11 is Gemini South, which is basically LSST as far as positions on the Earth are concerned. X05 probably didn't exist as a code when this was written (it was just designated last year, I believe), but as far as science is concerned these are all equivalent except in very extreme cases (impactors). 

I've created DM-45735, and will cite it in comments! Should be resolved soon, but who knows.